prov/rxm/src/rxm_attr.c

```diff
@@ -38,13 +38,16 @@
 
 #define RXM_DOMAIN_CAPS (FI_LOCAL_COMM | FI_REMOTE_COMM)
 
+// TODO have a separate "check info" against which app hints would be checked.
+
 /* Since we are a layering provider, the attributes for which we rely on the
  * core provider are set to full capability. This ensures that ofix_getinfo
  * check hints succeeds and the core provider can accept / reject any capability
  * requested by the app. */
 
 struct fi_tx_attr rxm_tx_attr = {
 	.caps = RXM_EP_CAPS,
+	.op_flags = ~0ULL,
```

Comment:
I don't believe you can just pass op_flags through.  Some op flags require that it be handled at the rxm level, and passing those through to the core provider can fail the request, when it should succeed.  Some sort of filtering is needed on the op_flags requested by the app to those which should pass through, and a check is needed to verify that rxm either handles a flag or can safely pass it through.

E.g. at least FI_MULTI_RECV and FI_COMPLETION are intercepted by rxm and not passed through.  There are likely others (FI_MORE, FI_INJECT, FI_DELIVERY_COMPLETE).

Reply:
`rxm_info_to_core` function filters flags like FI_MULTI_RECV when the hints is passed to core provider. The attributes here are used for checking app hints. So we need to fake support for all capabilities so that it passes `ofi_check_info` at rxm. I'm planning to have an explicit rxm_check_info to improve readability in a follow-up PR.

Isn't FI_MORE (more requests follow current one) applicable only for individual requests? I don't see it in fi_endpoint man page.

Reply:
FI_MORE isn't valid here, but FI_MULTICAST is, and rxm doesn't support that.  rxm should only set the flags that it knows how to handle, including those handled by passing through.  As soon as we define a new valid flag (e.g. FI_COLLECTIVE), this code has a high chance of breaking.  This likely shouldn't differ too much from how RXM_EP_CAPS is handled.

Reply:
So the correct way to check is to define pass through attributes for all capabilities, flags, etc and use it to filter app hints. That's what I am leaning towards in the follow up PR. I'll update in this PR just for op_flags.

Reply:
Updated the PR to set op_flags in "check_info" correctly.